applications/source/kafka-source/README.adoc

```diff
@@ -0,0 +1,29 @@
+//tag::ref-doc[]
+= Apache Kafka Source
+
+This module consumes messages from Apache Kafka.
+
+== Options
+
+The **$$kafka$$** $$source$$ has the following options:
+
+(See the Spring Boot documentation for Spring for Apache Kafka configuration properties)
+
+//tag::configuration-properties[]
+Properties grouped by prefix:
+
+
+=== kafka.supplier
+
+$$ack-discarded$$:: $$Whether to acknowledge discarded records after 'RecordFilterStrategy'.$$ *($$Boolean$$, default: `$$false$$`)*
+$$topic-pattern$$:: $$Apache Kafka topics pattern to subscribe.$$ *($$Pattern$$, default: `$$<none>$$`)*
+$$topics$$:: $$Apache Kafka topics to subscribe.$$ *($$String[]$$, default: `$$<none>$$`)*
+
+=== spring.kafka
+
+$$bootstrap-servers$$:: $$Comma-delimited list of host:port pairs to use for establishing the initial connections to the Kafka cluster. Applies to all components unless overridden.$$ *($$List<String>$$, default: `$$<none>$$`)*
+$$client-id$$:: $$ID to pass to the server when making requests. Used for server-side logging.$$ *($$String$$, default: `$$<none>$$`)*
+$$properties$$:: $$Additional properties, common to producers and consumers, used to configure the client.$$ *($$Map<String, String>$$, default: `$$<none>$$`)*
+//end::configuration-properties[]
```

Comment:
Pay attention, please, to this content from a `KafkaProperties`.
It is definitely not the whole list of props presented over there.
Not sure what is going on, but probably some issue in the doc generation plugin.

Reply:
I don't think that Christian's changes in the apps plugin have any effect on a docs plugin.
I think the way  have it fixed now with nested classes in that our properties file is correct.